Branch-SDK/BNCSKAdNetwork.m

```diff
@@ -91,4 +101,94 @@ - (void)updatePostbackConversionValue:(NSInteger)conversionValue
     
 }
 
+- (void)updatePostbackConversionValue:(NSInteger)fineValue
+                          coarseValue:(NSString *)coarseValue
+                           lockWindow:(BOOL)lockWindow
+                    completionHandler:(void (^)(NSError *error))completion {
+    if (@available(iOS 16.1, *)) {
+        if ([self shouldAttemptSKAdNetworkCallout]) {
+            
+            ((id (*)(id, SEL, NSInteger, NSString *, BOOL, void (^)(NSError *error)))[self.skAdNetworkClass methodForSelector:self.skAdNetworkUpdatePostbackConversionValueCoarseValueLockWindow])(self.skAdNetworkClass, self.skAdNetworkUpdatePostbackConversionValueCoarseValueLockWindow, fineValue, coarseValue, lockWindow, completion);
+        }
+    }
+}
+
+- (int) calculateSKANWindowForTime:(NSDate *) currentTime{
+    
+    int firstWindowDuration = 2 * 24 * 3600;
+    int secondWindowDuration = 7 * 24 * 3600;
+    int thirdWindowDuration = 35 * 24 * 3600;
+    
+    NSTimeInterval timeDiff = [currentTime timeIntervalSinceDate:self.installDate];
```

Comment:
@NidhiDixit09 to confirm self.installDate is currentInstallTimestamp and not first install timestamp correct?

Reply:
agreed, this makes sense!